TLM/TLM/Util/ModsCompatibilityChecker.cs

```diff
@@ -1,82 +1,201 @@
+            Log.Info("Scanning for incompatible mods");
+
+            // list of installed incompatible mods
+            Dictionary<PluginInfo, string> results = new Dictionary<PluginInfo, string>();
+
+            // only check enabled mods?
+            bool filterToEnabled = State.GlobalConfig.Instance.Main.IgnoreDisabledMods;
+
+#if !DEBUG
+            bool offline = IsOffline();
+#endif
 
-            if (incompatibleMods.Count > 0) {
-                Log.Warning("Incompatible mods detected! Count: " + incompatibleMods.Count);
+            // iterate plugins
+            foreach (PluginInfo mod in Singleton<PluginManager>.instance.GetPluginsInfo())
+            {
+                if (!mod.isBuiltin && !mod.isCameraScript && (!filterToEnabled || mod.isEnabled))
+                {
+                    string modName = GetModName(mod);
 
-                if (State.GlobalConfig.Instance.Main.ScanForKnownIncompatibleModsAtStartup) {
-                    IncompatibleModsPanel panel = UIView.GetAView().AddUIComponent(typeof(IncompatibleModsPanel)) as IncompatibleModsPanel;
-                    panel.IncompatibleMods = incompatibleMods;
-                    panel.Initialize();
-                    UIView.PushModal(panel);
-                    UIView.SetFocus(panel);
+                    if (incompatibleMods.ContainsKey(mod.publishedFileID.AsUInt64))
+                    {
+                        Log.Info($"Incompatible mod: {mod.publishedFileID.AsUInt64} - {modName}");
+                        results.Add(mod, modName);
+                    }
+#if !DEBUG
+                    // Workshop TM:PE builds treat local builds as incompatible
+                    else if (!offline && mod.publishedFileID.AsUInt64 == LOCAL_MOD && (modName.Contains("TM:PE") || modName.Contains("Traffic Manager")))
+                    {
+                        Log.Info($"Local TM:PE detected: '{modName}' in '{mod.modPath}'");
+                        string folder = mod.modPath.Split(Path.DirectorySeparatorChar).Last();
+                        results.Add(mod, $"{modName} in /{folder}");
+                    }
+#endif
                 }
-            } else {
-                Log.Info("No incompatible mods detected");
             }
+
+            Log.Info($"Scan complete: {results.Count} incompatible mod(s) found");
+
+            return results;
+        }
+
+        /// <summary>
+        /// Gets the name of the specified mod.
+        /// 
+        /// It will return the <see cref="IUserMod.Name"/> if found, otherwise it will return <see cref="PluginInfo.name"/> (assembly name).
+        /// </summary>
+        /// 
+        /// <param name="plugin">The <see cref="PluginInfo"/> associated with the mod.</param>
+        /// 
+        /// <returns>The name of the specified plugin.</returns>
+        public string GetModName(PluginInfo plugin)
+        {
+            string name = plugin.name;
+            IUserMod[] instances = plugin.GetInstances<IUserMod>();
```

Comment:
Use `plugin.userModInstance` instead.

Reply:
I'm not sure what I'm doing wrong, but it's complaining at this:

```csharp
public string GetModName(PluginInfo plugin)
{
        return plugin.userModInstance.Name;
}
```

Says `Cannot convert method group 'Name' to non-delegate type 'string'. Did you intend to invoke the method?`

Which is weird, because I thought it was treated like a getter? So then I try:

> return plugin.userModInstance.Name();

Now I get `The type 'object' must be non-nullable value type in order to use it as parameter 'T' in the generic type or method 'Enum.extensions.Name<T>(T)`

I have no idea what that means.

Reply:
You have to cast first:
```
((IUserMod)plugin.userModInstance).Name
```
The `userModInstance` property is of type `object` (actually, untyped). Using `Name` on an `object` resolves to some extension method `'Enum.extensions.Name<T>(T)` in your case (no idea where this method comes from, maybe it's Unity or old TM:PE code).

Reply:
That works, thanks! Still trying to wrap my head around the whole typecasting thing.